R/req-options.R

```diff
@@ -49,12 +49,12 @@ req_user_agent <- function(req, string = NULL) {
   check_request(req)
 
   if (is.null(string)) {
-    versions <- c(
-      httr2 = utils::packageVersion("httr2"),
-      `r-curl` = utils::packageVersion("curl"),
-      libcurl = curl::curl_version()$version
+    versions <- list(
+      httr2 = packageVersion("httr2"),
+      `r-curl` = packageVersion("curl"),
+      libcurl = curl_version()$version
     )
-    string <- paste0(names(versions), "/", versions, collapse = " ")
+    string <- paste0(names(versions), "/", vapply(versions, as.character, ""), collapse = " ")
```

Comment:
I'd put the `as.character()` around the `packageVersion()` calls to make it more clear where the problem comes from.

Reply:
OK, switched